test/dataframe.jl

```diff
@@ -452,7 +452,14 @@ module TestDataFrame
     df = DataFrame(A = 1:10, B = 'A':'J')
     @test !(df[:,:] === df)
 
-    @test append!(DataFrame(A = 1:2, B = 1:2), DataFrame(A = 3:4, B = 3:4)) == DataFrame(A=1:4, B = 1:4)
+    df = DataFrame(A = 1:2, B = 1:2)
+    df2 = DataFrame(A=1:4, B = 1:4)
+    @test append!(df, DataFrame(A = 3:4, B = [3.0, 4.0])) == df2
+    @test_throws InexactError append!(df, DataFrame(A = 3:4, B = [3.5, 4.5]))
+    @test df == df2
+    @test_throws MethodError append!(df, DataFrame(A = 3:4, B = ["a", "b"]))
+    @test df == df2
```

Comment:
Would be good to add a test which includes `missing`:
```julia
using DataFrames

df = DataFrame(A=[missing, 1])
df2 = DataFrame(A=3:4)
append!(df, df2)
```

Reply:
There's really nothing special about that example, `Int` is a subtype of `Union{Int,Missing}`. So I don't really see why we'd need to test it.